html-report/src/components/SearchBar.js

```diff
@@ -4,30 +4,47 @@ import cx from 'classnames';
 import elasticlunr from 'elasticlunr';
 import convertTime from './../utils/convertTime'
 
-const SEARCH_FIELDS = ['package_name', 'class_name', 'name', 'id', 'status'];
-const SEARCH_REF = 'id';
 const EL_SEARCH = elasticlunr();
 export default class SearchBar extends Component {
   static propTypes = {
-    setSearchResults: PropTypes.func
+    setSearchResults: PropTypes.func,
+    setPerformFilterSearchCallback: PropTypes.func,
+    searchFields: PropTypes.array,
+    searchRef: PropTypes.string,
+    data: PropTypes.array,
   };
 
   state = {
-    data: window.suite.tests,
+    data: this.props.data,
     error: false,
     searchLabel: null,
     searchParams: null,
     query: ''
   };
 
   componentWillMount() {
-    let { data } = this.state;
     elasticlunr.clearStopWords();
-    SEARCH_FIELDS.forEach(f => EL_SEARCH.addField(f))
-    EL_SEARCH.setRef(SEARCH_REF);
-    if (data.length) {
-      data.forEach(item => EL_SEARCH.addDoc(item))
-    }
+    this.props.searchFields.forEach(f => EL_SEARCH.addField(f))
+    EL_SEARCH.setRef(this.props.searchRef);
+  }
+
+  componentDidMount() {
+    this.props.setPerformFilterSearchCallback(this.performFilterSearch)
+  }
+
+  componentWillReceiveProps(props) {
+      let data = props.data;
+      if (data === this.state.data) {
```

Comment:
This code will always result in `false`, please read the JS documentation on how arrays works.